### DISCUSSION TITLE
PointAnnotation not loading icons/markers, MarkerView doesn't have click event even if i ad touchable opacity it does not work can some help!!

### DISCUSSION THREAD

**Discussion by sabinsthnp** [rnmapbox/maps#2833] [Q&A]
const Marker = ({ selectedCordinates,coordinate, image, id,data}) => {
  console.log("🚀 ~ file: MapBox.js:74 ~ Marker ~ selectedCordinates:", selectedCordinates,coordinate)
  const dispatch = useDispatch();
  
  const componentRef = useRef(null)
  const clickHandle = () => {
    console.log(":::Opening Details:::");
    navigate(dashboard, { page: PROSHOP, passedData: {item:data.data}})
    dispatch(setSelectedCoordinates([parseInt(coordinate.longitude),parseInt(coordinate.latitude)]))
  };
  return (
      <MapboxGL.PointAnnotation
        // centerCoordinate={coordinate}
        // key="PointAnnotation"
        id={`place-${id}`}
        isSelected={true}
        title="Test"
        anchor={{ x: 0.5, y: 1.0 }}
        // key={`place-${data?.data?.vendor_id || data?.data?.id}`}
        // draggable
        // onDragEnd={(data)=>{console.log("drag end",data);}}
        coordinate={coordinate}
       onSelected={clickHandle}
      >

        <View
          style={
            {
              height: 30,
              width: 80,
              // backgroundColor: "red",
              borderRadius: 50,
              borderColor: "#fff",
              borderWidth: 0,
              display:'flex',
              flexDirection:'row',
              alignContent:'center',
              alignSelf:'center'
            }
          }
        >
          {selectedCordinates[0]!=coordinate[0] ?
               <WithLocalSvg asset={svgFileS} width={30} height={30} />
               : <WithLocalSvg asset={svgFileSR} width={40} height={40} />
              }
          <Text style={{fontSize:Fonts.verysmall,color:Colors.black,fontWeight:Fonts.bold}}> {data?.data?.store_name} {data?.data?.patient} </Text>
          

        </View>
      </MapboxGL.PointAnnotation>
  );
};

**Comment by stevenharderjr** [comment]
I wouldn't expect PointAnnotation to work outside of a MapView.

**Reply by sabinsthnp** [reply]
this is a component of a marker used inside mapview

**Comment by MaxKicw** [comment]
Hi mate, it's a bit late now but I found a workaround for this. The `onPressIn()` and the `onPressOut() `methods are working. 
onPress() is also not working for me. Hope it helps you or some reader :)
